src/internal/connector/onedrive/item.go

```diff
@@ -61,18 +62,40 @@ func oneDriveItemMetaReader(
 	service graph.Servicer,
 	driveID string,
 	item models.DriveItemable,
+	fetchPermissions bool,
 ) (io.ReadCloser, int, error) {
-	meta, err := oneDriveItemMetaInfo(ctx, service, driveID, item)
-	if err != nil {
-		return nil, 0, err
+	meta := Metadata{
+		FileName: *item.GetName(),
 	}
 
-	metaJSON, err := json.Marshal(meta)
+	perms, err := oneDriveItemPermissionInfo(ctx, service, driveID, item, fetchPermissions)
 	if err != nil {
+		// Keep this in an if-block because if it's not then we have a weird issue
+		// of having no value in error but golang thinking it's non nil because of
+		// the way interfaces work.
+		err = clues.Wrap(err, "fetching item permissions")
+	} else {
+		meta.Permissions = perms
+	}
+
+	metaJSON, serializeErr := json.Marshal(meta)
```

Comment:
Should we bother trying to marshall if we hit an error in 76?

Reply:
that's a good question. I think it would be best if we do because then we have the highest change of recovering some data down the line even if we had issues. Since the file name is now in the metadata file it's harder for us to ensure we have all the things we need to restore an item properly

If we skip marshalling data then there's the chance we'll end up with the item data but not the item name (depending on how fail-fast semantics evolve as well). If we do marshall it then maybe we won't have permissions, but we'll at least have the file name